pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -200,3 +297,143 @@ func (k *klusterletStatusController) sync(ctx context.Context, controllerContext
 	)
 	return nil
 }
+
+func buildKubeClientWithSecret(secret *corev1.Secret) (kubernetes.Interface, error) {
+	defer func() {
+		for key := range secret.Data {
+			os.Remove(path.Join("/tmp", key))
+		}
+	}()
+
+	for key, data := range secret.Data {
+		if err := ioutil.WriteFile(path.Join("/tmp", key), data, 0644); err != nil {
+			return nil, err
+		}
+	}
+	restConfig, err := clientcmd.BuildConfigFromFlags("", path.Join("/tmp", "kubeconfig"))
```

Comment:
Instead of writing to the filesystem, use the [`Load`](https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/loader.go#L406) method, which takes a byte array and gives you a config.

Reply:
I think we cannot use `Load` method here, for hub `kubeconfig`, its `client-certificate` and `client-key` are file path, rather than byte array, so if we use `Load` method, it will be failed, also for bootstrap `kubeconfig`, it is mounted from a secret, I think it also may be the file path, rather than the byte array

Reply:
Okay - let's use ioutil.TempDir so that these tests don't collide with each other then

Reply:
done